### ISSUE TITLE
Restrictions on foreign keys

### ISSUE THREAD

**Issue by johnnncodes** [mickhansen/ssacl-attribute-roles#5]
Given this User model:

```
var ssaclAttributeRoles = require('ssacl-attribute-roles');

module.exports = function(sequelize, DataTypes) {
  var Model = sequelize.define("User", {
    password: {
      type: DataTypes.STRING,
      roles: false
    },
  }, {
    classMethods: {
      associate: function(models) {
        Model.belongsTo(models.Company);
      }
    }
  });

  ssaclAttributeRoles(Model);

  return Model;
};
```

How can I restrict CompanyId in Users table to be updated? Is this currently possible?

**Comment by mickhansen** [comment]
While still allowing the use of association setters?
That is probably not possible at the moment, depends on the Sequelize internals (whether they use `raw: true` for setting the value, which they probably should).

/cc @janmeier

**Comment by janmeier** [comment]
Pretty sure they don't right now, but they probably should

**Comment by johnnncodes** [comment]
@mickhansen Yes :) Anyway, thanks for the response. I'll just restrict it manually for now.

**Comment by CrushedPixel** [comment]
Any update on this? Does Sequelize internally use `raw: true` nowadays?

**Comment by mickhansen** [comment]
@CrushedPixel Library not maintained, no update.

**Comment by CrushedPixel** [comment]
Does this refer to ssacl-attribute-roles or Sequelize itself? If I understood correctly, Sequelize would need to use `raw: true`, not ssacl-attribute-roles. Is Sequelize seriously unmaintained?

**Comment by mickhansen** [comment]
Obviously i mean this library
